src/sensor_ads1220.c

```diff
@@ -0,0 +1,206 @@
+// Support for ADS1220 ADC Chip
+//
+// Copyright (C) 2024 Gareth Farrington <gareth@waves.ky>
+//
+// This file may be distributed under the terms of the GNU GPLv3 license.
+
+#include "board/irq.h" // irq_disable
+#include "board/gpio.h" // gpio_out_write
+#include "board/misc.h" // timer_read_time
+#include "basecmd.h" // oid_alloc
+#include "command.h" // DECL_COMMAND
+#include "sched.h" // sched_add_timer
+#include "sensor_bulk.h" // sensor_bulk_report
+#include "spicmds.h" // spidev_transfer
+
+struct ads1220_adc {
+    struct timer timer;
+    uint32_t rest_ticks;
+    struct gpio_in data_ready;
+    struct spidev_s *spi;
+    uint8_t flags, data_count;
+    struct sensor_bulk sb;
+};
+
+// Flag types
+enum {
+    FLAG_PENDING = 1 << 0
+};
+
+#define BYTES_PER_SAMPLE 4
+
+static struct task_wake wake_ads1220;
+
+/****************************************************************
+ * ADS1220 Sensor Support
+ ****************************************************************/
+#define MAX_SPI_READ_TIME timer_from_us(150)
+
+static inline uint8_t
+is_flag_set(const uint8_t mask, struct ads1220_adc *ads1220)
+{
+    return !!(mask & ads1220->flags);
+}
+
+static inline void
+set_flag(uint8_t mask, struct ads1220_adc *ads1220)
+{
+    ads1220->flags |= mask;
+}
+
+static inline void
+clear_flag(uint8_t mask, struct ads1220_adc *ads1220)
+{
+    ads1220->flags &= ~mask;
+}
```

Comment:
Seems like dead code.  is_flag_set() and clear_flag() are never used.  Nothing ever clears the flag, so set_flag() is redundant..

Reply:
Yep, I think I refactored this out but never deleted them, will fix.

Reply:
Removed